### ISSUE TITLE
GAN Loss

### ISSUE THREAD

**Issue by dingyan1478** [JunlinHan/DCLGAN#1]
Hello!
I want to know why use hinge as  gan loss ?
minvalue = torch.min(prediction - 1, torch.zeros(prediction.shape).to(prediction.device))
The network output of discriminator D is not limited to 0-1, so I want to know the reason for using this loss function. If a pixel value in the prediction value is 1.03 after D, then the output of minvalue at this time is still 0, and the prediction result at this time is not correct, right?
Maybe there are some questions in my understanding. I sincerely look forward to your reply!

**Comment by JunlinHan** [comment]
Hi dingyan,

From my understanding, hinge gan sets a threshold to smooth the training.

For example, 
if target_is_real:
minvalue = torch.min(prediction - 1, torch.zeros(prediction.shape))
loss = -torch.mean(minvalue),

If the prediction is larger than 1, hinge gan think it's too large, and set the loss to 0, to prevent the discriminator from "over" optimization.
Hope it helps.

**Comment by dingyan1478** [comment]
Thank you!
